helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ParticipantMessageMonitor.java

```diff
@@ -19,86 +19,93 @@
  * under the License.
  */
 
-public class ParticipantMessageMonitor implements ParticipantMessageMonitorMBean {
+import java.util.ArrayList;
+import java.util.List;
+import javax.management.JMException;
+import javax.management.ObjectName;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMBeanProvider;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.SimpleDynamicMetric;
+
+
+public class ParticipantMessageMonitor extends DynamicMBeanProvider {
   public static final String PARTICIPANT_KEY = "ParticipantName";
   public static final String PARTICIPANT_STATUS_KEY = "ParticipantMessageStatus";
 
+  // For registering dynamic metrics
+  private final ObjectName _initObjectName;
+  private final String _participantName;
+
+  private SimpleDynamicMetric<Long> _receivedMessages;
+  private SimpleDynamicMetric<Long> _discardedMessages;
+  private SimpleDynamicMetric<Long> _completedMessages;
+  private SimpleDynamicMetric<Long> _failedMessages;
+  private SimpleDynamicMetric<Long> _pendingMessages;
+
   /**
    * The current processed state of the message
    */
   public enum ProcessedMessageState {
-    DISCARDED,
-    FAILED,
-    COMPLETED
+    DISCARDED, FAILED, COMPLETED
```

Comment:
NIT: I think our formatter has problem. Let manually fix this first.

Reply:
Yeah, seems caused by formatter. Changed back.